dhis-2/dhis-services/dhis-service-analytics/src/test/java/org/hisp/dhis/analytics/common/DimensionIdentifierConverterTest.java

```diff
@@ -93,11 +94,11 @@ void fromStringWithSuccessNoOffset()
         final String fullDimensionId = "lxAQ7Zs9VYR.RaMbOrTys0n.jklm";
 
         // When
-        final DimensionIdentifier<Program, ProgramStage, String> dimensionIdentifier = new DimensionIdentifierConverter()
+        final DimensionIdentifier<Program, ProgramStage, StringUid> dimensionIdentifier = new DimensionIdentifierConverter()
             .fromString( programs, fullDimensionId );
 
         // Then
-        assertEquals( "jklm", dimensionIdentifier.getDimension(), "Dimension uid should be jklm" );
+        assertEquals( "jklm", dimensionIdentifier.getDimension().getUid(), "Dimension uid should be jklm" );
         assertEquals( "lxAQ7Zs9VYR", dimensionIdentifier.getProgram().getElement().getUid(),
             "Program uid should be lxAQ7Zs9VYR" );
         assertEquals( null, dimensionIdentifier.getProgram().getOffset(),
```

Comment:
I know it is old code, but can you use on line 104 assertNull? The same on lines 108, 133, 135, 157, 158